sdk/identity/Azure.Identity/tests/samples/TokenCacheSnippets.cs

```diff
@@ -12,37 +12,55 @@ namespace Azure.Identity.Samples
 {
     public class TokenCacheSnippets
     {
+        #region Snippet:Identity_TokenCache_CustomPersistence_Usage_TokenCachePath
```

Comment:
Good catch.  I updated the md.

Reply:
For disposal, yes, it will fail if you write to the file and immediately read from it in the same scope.  This was uncovered when I was demonstrating its usage to a customer.  You are right that it is not apparent in this sample, because it is not immediately read.  In general, I avoid using the using declaration because I have run into scoping issues like this in the past.  I look at it the same way I do if we should include "{}" on each conditional.  I do it because I have screwed that up in the past, didn't include the curly's, and then the conditional breaks when I need to modify it.

I'm okay with changing back or expanding the sample if you think it helps.

Reply:
I think it makes sense as a good best practice, I was more curious how the issue would manifest. I'm fine with keeping it with the braces, but if you think there is a way to demonstrate the scenario where reading the file afterwards that is relevant to the sample, I think it would be nice.

Reply:
I personally would rather not update the sample to reflect that.  Let's see what @schaabs says about this.

Reply:
I don't think that writing the file and and then reading it in the same scope would be a common scenario, so I don't think was should demonstrate doing this in a snippet. However, I think including the braces here makes these snippets more "portable", in that users can paste them into their code and be guaranteed that the file is written after the snippet executes.

Reply:
Okay, resolving this one.  Thanks for discussion.